src/formatDate.js

```diff
@@ -50,7 +50,50 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    const part = toFormat[i];
+    const index = fromFormat.indexOf(part);
+
+    newDate[i] = dateArray[index];
+
+    if (!newDate[i]) {
+      newDate[i] = getYear(dateArray, fromFormat, toFormat);
+    }
+  }
+
+  return newDate.join(toFormat[3]);
+}
+
+function getYear(dateArray, fromFormat, toFormat) {
+  let indexFromYear;
+  let indexToYear;
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      indexFromYear = i;
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      indexToYear = i;
+    }
+  }
+
+  let year = dateArray[indexFromYear];
+
+  if (fromFormat[indexFromYear] === 'YYYY'
+    && toFormat[indexToYear] === 'YY') {
+    year = year.substring(2);
+  }
+
+  if (fromFormat[indexFromYear] === 'YY'
+    && toFormat[indexToYear] === 'YYYY') {
+    year = (year < 30) ? '20' + year : '19' + year;
+  }
+
+  return year;
```

Comment:
```suggestion
  if (fromFormat[indexFromYear] === toFormat[indexToYear]) {
    return year;
  }

  if (fromFormat[indexFromYear] === 'YYYY') { // after first if fromYear and toYear are not equal
    return year.substring(2);
  }
    // fromFormat is 'YY' and toFormat is 'YYYY', 'cause of previous conditions 
    
    return (year < 30)   // wrap ternary operators conditions to new line, it's more readable
      ? '20' + year
      : '19' + year;  
```

Reply:
Conditions in if looks complicated because of their length, let's make it simpler